plugins/modules/inventory_intent.py

```diff
@@ -2804,9 +2882,9 @@ def get_diff_merged(self, config):
                     device_id = self.get_device_ids([device_ip])
                     device_role_args = self.config[0].get('update_device_role')
 
-                    if 'role' not in device_role_args or 'role_source' not in device_role_args:
+                    if 'role' not in device_role_args:
                         self.status = "failed"
-                        self.msg = "Mandatory paramter(role/sourceRole) to update Device Role are missing"
+                        self.msg = "Mandatory paramter(role) to update Device Role is missing"
```

Comment:
spelling mistake.. 
self.msg = "Mandatory parameter (role) to update Device Role is missing."